### DISCUSSION TITLE
libudev required for SDL2 build?

### DISCUSSION THREAD

**Discussion by MichaIng** [BlitterStudio/amiberry#846] [Q&A] [answered]
Based on the older Wiki instructions for building SDL2 it says that `libudev-dev` needs to be installed, which implies the `libudev1` dependency for the final Amiberry build. Does anyone know why or for what this is required? It is generally an optional feature which can be enabled and disabled via configure flags when building SDL2, but I have no idea which additional and probably useful or required features it enables for Amiberry.

**Comment by midwan** [comment]
Hm, good question. I'm not really sure if it's needed actually, I think it was included because SDL2 seemed to use it - I don't remember if there were any problems if it was missing. 

I guess the best way is to try and find out. Worst case scenario is that something doesn't work. :)

**Comment by MichaIng** [comment, marked as answer]
The SDL2 source says "for device enumeration and hotplugging" and it appears in combination with joystick, gamepad, hidapi and evdev source files 🤔. Sadly nothing I can test here 😄. I mean it is not a big issue, as `udev` provides essential system features and hence is present anyway where relevant to us, but would be still interesting it Amiberry, respectively the games do make use of it.

**Reply by midwan** [reply]
I guess one way to test it would be to see if it still detects a keyboard/joystick without it. ;)

**Reply by MichaIng** [reply]
Especially also whether it detects it when being hot plugged. Sadly I don't have either of it. When finding time, I'll try it with mouse and keyboard.
